iohub/dataset_base.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from pathlib import Path
+from types import TracebackType
+from typing import Any, Iterable, Optional, Type, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+_AXES_PREFIX = ["T", "C", "Z", "Y", "X"]
+
+
+class BaseFOV(ABC):
+    _root: Path
+
+    @property
+    @abstractmethod
+    def axes_names(self) -> list[str]:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def channels(self) -> list[str]:
```

Comment:
Maybe `channels` -> `channel_names`, since `channels` might make you think you're getting the number of channels?